lib/note-content-editor.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { PropTypes } from 'react';
+import { Editor, EditorState, ContentState } from 'draft-js';
+import { urlDecorator } from './url-decorator';
+
+function plainTextContent( editorState ) {
+	return editorState.getCurrentContent().getPlainText( '\n' )
+}
+
+export default React.createClass( {
+
+	propTypes: {
+		content: PropTypes.string.isRequired,
+		onChangeContent: PropTypes.func.isRequired
+	},
+
+	getInitialState() {
+		const { content } = this.props;
+		const contentState = ContentState.createFromText( content, '\n' );
+		return {
+			editorState: EditorState.createWithContent( contentState, urlDecorator )
+		}
+	},
+
+	handleEditorStateChange( editorState ) {
+		const contentHasChanged = plainTextContent( editorState )
+			!== plainTextContent( this.state.editorState )
```

Comment:
I haven't done that but for a different reason — `contentState` (despite its name, this is actually not great naming on the part of `draft-js` I guess, see https://facebook.github.io/draft-js/docs/api-reference-content-state.html#content) contains info about both content and selection state.